src/components/OptimizedImage.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+export default function OptimizedImage(props) {
+  const {
+    src,
+    width = 320,
+    height,
+    quality,
+    screenWidths = [1600, 1280, 960, 0],
+    imageSizes = [400, 300, 250, 200],
+    fixed,
+    ...rest
+  } = props;
+
+  if (!src) return <></>;
+
+  const cdnPath = 'https://cdn.statically.io/img';
+  const domain = src.match(/\/\/(.*)\//)[1];
+  const imagePath = src.match(/.com\/(.*)/)[1];
```

Comment:
I think this is a reasonable assumption, but it would also be good to fall back to the original image if we can't match the image path or domain.

You could combine and generalise these two checks to:
```js
const matches = src.match(/\/\/(.*)\/(.*)/);
if (matches.length > 2) {
  const domain = matches[1];
  const imagePath = matches[2];
  ...
}
```

Reply:
Thanks for the great feedback @nmcharlton! I made the necessary changes.